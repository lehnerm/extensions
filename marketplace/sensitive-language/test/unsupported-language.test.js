'use strict';

const React = require('react');
const renderer = require('react-test-renderer');

function mockComponent (componentClassName) {
  return ((props) => {
    const className = props.className ? `${componentClassName} ${props.className}` : componentClassName;

    return (<div className={className} {...props}>{props.children}</div>)
  });
}

jest.mock('@contentful/forma-36-react-components', () => ({
  Icon: mockComponent('icon'),
  Typography: mockComponent('typography'),
  Subheading: mockComponent('subheading'),
  Paragraph: mockComponent('paragraph')
}));

const {UnsupportedLanguage} = require('../src/unsupported-language.js');

test('UnsupportedLanguage: renders correctly', function () {
  const tree = renderer
  .create(<UnsupportedLanguage localeCode="de-De" localeName="German" />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
