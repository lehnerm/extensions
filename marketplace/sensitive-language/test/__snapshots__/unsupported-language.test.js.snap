// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnsupportedLanguage: renders correctly 1`] = `
<div
  className="typography"
>
  <div
    className="subheading"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
      }
    }
  >
    <div
      className="icon"
      color="negative"
      extraClassNames="f36-margin-right--xs"
      icon="Warning"
    />
    Unsupported language
  </div>
  <div
    className="paragraph"
    style={
      Object {
        "marginBottom": 0,
      }
    }
  >
    The sensitive language feature only works for English text. The default locale of this space is German (de-De).
  </div>
</div>
`;
